### ISSUE TITLE
Cannot share file from Safari

### ISSUE THREAD

**Issue by m1st1ck** [achorein/expo-share-intent#104]
**Describe the bug**
In Safari if the web page is a file and we try to share it, our app is not showing as an option. We are able to share images, videos, files, audio from Photos and Files, also regular web pages and selected text. Android works fine and so does iOS chrome.

Here is our plugin config. Maybe we are missing some activation rule, but could not find information.
```
 "plugins": [
      [
        "expo-share-intent",
        {
          "iosActivationRules": {
            "NSExtensionActivationSupportsText": true,
            "NSExtensionActivationSupportsWebURLWithMaxCount": 1,
            "NSExtensionActivationSupportsWebPageWithMaxCount": 1,
            "NSExtensionActivationSupportsImageWithMaxCount": 10,
            "NSExtensionActivationSupportsMovieWithMaxCount": 1,
            "NSExtensionActivationSupportsFileWithMaxCount": 1
          },
          "androidIntentFilters": [
            "text/*",
            "image/*",
            "video/*",
            "audio/*",
            "application/*"
          ],
          "androidMultiIntentFilters": [
            "image/*"
          ]
        }
      ]
    ]
```

https://github.com/user-attachments/assets/55b82f24-32b3-416a-b11a-24f84de63db3

**To Reproduce**
- create new expo project `npx create-expo-app StickerSmash --template blank`
- install and configure `expo-share-intent`
- open http://www.labottega.bg/assets/winelist_apr2023.pdf in safari and try to share it

**Additional context**
- expo version : 51
- using firebase : no
- package version : 2.3.0
- platform target : iOS
- device : simulator & real device
- type: local build

**Comment by achorein** [comment]
When you initiate a share for a PDF in Safari, it will actually consider 2 input items: the PDF and the URL. To handle this we need a custom `NSExtensionActivationRule`. But it's not handle in this plugin yet.

a workaround  is to use chrome on iOS to share pdf from the url

you can configure custom activation rule like this on your `app.json`:
```
    "plugins": [
      [
        "expo-share-intent",
        {
              "iosActivationRules": "SUBQUERY (\nextensionItems,\n$extensionItem,\nSUBQUERY (\n$extensionItem.attachments,\n$attachment,\nANY $attachment.registeredTypeIdentifiers UTI-CONFORMS-TO \"com.adobe.pdf\"\n|| ANY $attachment.registeredTypeIdentifiers UTI-CONFORMS-TO \"public.file-url\"\n|| ANY $attachment.registeredTypeIdentifiers UTI-CONFORMS-TO \"public.url\"\n|| ANY $attachment.registeredTypeIdentifiers UTI-CONFORMS-TO \"public.jpeg\"\n|| ANY $attachment.registeredTypeIdentifiers UTI-CONFORMS-TO \"public.png\"\n).@count == $extensionItem.attachments.@count\n).@count >= 1",
         ...
        }
      ],
```

with this configuration (and some code update), the app appear in sharing view.

<img width="360" alt="image" src="https://github.com/user-attachments/assets/7478d8a7-92a0-4f2f-ab5d-fce2c2a17eca">

but the plugin doesn't handle it in the custom view

<img width="249" alt="image" src="https://github.com/user-attachments/assets/956ff3ab-f480-4f17-8c86-d12ca8b9d1f9">

**Comment by achorein** [comment]
@m1st1ck available in v2.5.0 using a custom activation rule !
